python/ray/rllib/optimizers/async_samples_optimizer.py

```diff
@@ -71,9 +76,12 @@ def __init__(self,
                 learner_queue_size=learner_queue_size,
                 _fake_gpus=_fake_gpus)
         else:
+            if old_worker:
+                weights = self.workers.local_worker().get_weights()
```

Comment:
does this have to happen here? or can we push it down to the learner thread?

Reply:
Yes, it should be moved to learner thread init method